glusterd2/commands/volumes/common.go

```diff
@@ -96,12 +96,14 @@ func expandGroupOptions(opts map[string]string) (map[string]string, error) {
 		if !ok {
 			options[opt] = val
 		} else {
-			for _, option := range optionSet {
+			for _, option := range optionSet.Options {
 				switch val {
 				case "on":
 					options[option.Name] = option.OnValue
 				case "off":
-					options[option.Name] = option.OffValue
+					if volinfo != nil {
+						delete(volinfo.Options, option.Name)
```

Comment:
This doesn't feel like the right place to do this. Now there shall be two places where `volinfo.Options` gets modified - once here and again later in the calling function.

`expandGroupOptions`, as the name says is meant to only expand the group options. Do not pass `Volinfo` here. Instead, you can find the default value of each option and explicitly set that in the `options` map. 

```go
case "off":
    op, err := xlator.FindOption(option.Name)
    if err != nil {
        return err
    }
    options[option.Name] = op.DefaultValue
```

A better place for doing `delete(volinfo.Options, option.Name)` would be in the volume `reset` operation when done on a group option. I don't think reset has been implemented yet.

Reply:
ok. But there should be some place, which filters the options if it has default value, so that vol info doesn't display the options as modified? From the user's perspective its really confusing to see the options even after setting <profile> off.

Reply:
> From the user's perspective its really confusing to see the options even after setting off

They can do a reset to not see those options.